articles/112_generated_interfaces_cpp.md

```diff
@@ -103,7 +103,19 @@ All other constants are declared as `static const` members in the struct and the
 
 ### Constructors
 
-The *default constructor* initializes all members with their default value.
+The *default constructor* initializes all members with their default value; if a field doesn't have a default value, then the field is [value-initialized](http://en.cppreference.com/w/cpp/language/value_initialization).
+In some cases this may not be desirable, since these fields will often be immediately overwritten with user-provided values.
+Therefore, the constructor takes an optional directive of type `enum rosidl_runtime_cpp_msg_init_type` to control how initialization is done:
+
+- `MSG_INIT_INITIALIZE_ALL` - Initialize all members with their default value; if a field doesn't have a default value, then the field is [value-initialized](http://en.cppreference.com/w/cpp/language/value_initialization).
+  - The safest option, and also the default (used if not passing any argument to the constructor).
+- `MSG_INIT_SKIP_INITIALIZE` - Don't initialize any members; it is the user's responsibility to ensure that all fields get initialized with some value, otherwise undefined behavior results
```

Comment:
> otherwise undefined behavior results

I don't think this is accurate. Undefined behavior doesn't necessarily happen - it could be though. I think it would be good to clarify that.

Reply:
Right.  Will clarify that point.